core/src/main/java/nl/nn/adapterframework/http/rest/ApiListenerServlet.java

```diff
@@ -358,14 +358,12 @@ private void handleRequest(HttpServletRequest request, HttpServletResponse respo
 				/*
 				 * Evaluate preconditions
 				 */
-				String acceptHeader = request.getHeader("Accept");
-				if(StringUtils.isNotEmpty(acceptHeader)) { // If an Accept header is present, make sure we comply to it!
-					if(!listener.accepts(acceptHeader)) {
-						response.setStatus(406);
-						response.getWriter().print("It appears you expected the MediaType ["+acceptHeader+"] but I only support the MediaType ["+listener.getContentType()+"] :)");
-						log.warn(createAbortMessage(remoteUser, 406) + "client expects ["+acceptHeader+"] got ["+listener.getContentType()+"] instead");
-						return;
-					}
+				final String acceptHeader = request.getHeader("Accept");
+				if(StringUtils.isNotBlank(acceptHeader) && !listener.accepts(acceptHeader)) { // If an Accept header is present, make sure we comply to it!
```

Comment:
Ik denk dat er in de buurt van deze code nog zoiets moet komen dat als de ApiListener een Produces=JSON heeft, en er geen Accept-Header is, de Json2XmlValidator dan toch die JSON door krijgt. Dat hij dan een Accept-Header injeteert in de context, bijvoorbeeld